scapy/layers/inet.py

```diff
@@ -216,6 +216,29 @@ class IPOption_Traceroute(IPOption):
                    IPField("originator_ip", "0.0.0.0")]
 
 
+class IPOption_Timestamp(IPOption):
+    name = "IP Option Timestamp"
+    optclass = 2
+    option = 4
+    fields_desc = [_IPOption_HDR,
+                   ByteField("length", None),
+                   ByteField("pointer", 9),
+                   BitField("oflw", 0, 4),
+                   BitEnumField("flg", 1, 4,
+                                {0: "timestamp_only",
+                                 1: "timestamp_and_ip_addr",
+                                 3: "prespecified_ip_addr"}),
+                   ConditionalField(IPField("internet_address", "0.0.0.0"),
+                                    lambda pkt: pkt.flg != 0),
+                   IntField('timestamp', 0)]
+
+    def post_build(self, p, pay):
+        if self.length is None:
+            length = len(p)
+            p = p[:1] + struct.pack("!B", length) + p[2:]
```

Comment:
I agree on this example with you @gpotter2 that it is better to add a specific `.post_build()`.

I think it is better to use the fact that we know the exact size, so that someone who reads the code immediately understands how the length may vary:
```suggestion
            p = p[:1] + struct.pack("!B", 8 + 4 * (self.flg != 0)) + p[2:]
```

And anyway, we don't need a variable for that:
```suggestion
            p = p[:1] + struct.pack("!B", len(p)) + p[2:]
```

Reply:
I've picked option 2. I find it the clearest